src/AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -20,19 +22,22 @@ public class GraphQLHttpMiddleware<TSchema> where TSchema : ISchema
         private readonly IDocumentExecuter _executer;
         private readonly IDocumentWriter _writer;
         private readonly TSchema _schema;
+        private readonly IEnumerable<IDocumentExecutionListener> _documentListeners;
 
         public GraphQLHttpMiddleware(
             RequestDelegate next,
             IOptions<GraphQLHttpOptions> options,
             IDocumentExecuter executer,
             IDocumentWriter writer,
-            TSchema schema)
+            TSchema schema,
+            IEnumerable<IDocumentExecutionListener> documentListeners)
```

Comment:
`IEnumerable<IDocumentExecutionListener> documentListeners)` should be moved to the method `Invoke` since the middleware is only instantiated one time at app startup.

For example DataLoaderDocumentListener is registrered as Transient in the [DataLoader tests](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.DataLoader.Tests/QueryTestBase.cs#L42). In this pull request the actual lifetime would instead be singleton.

See [Per-request dependencies](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?tabs=aspnetcore2x)

Reply:
Okey i will look into this. thanks!

Reply:
@pekkah @sahb1239 i have a heard time injecting it correct way to SubscriptionProtocolHandler.
i'm thinking of a factory or just inject the IServiceProvider to request service at execution.

Reply:
I think IServiceProvider would be sufficient

Reply:
Issue #56 is releated